src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -0,0 +1,167 @@
+package seedu.address.model.session;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+/**
+ * Class that handles RecurringSession that extend Session.
+ */
+public class RecurringSession extends Session {
```

Comment:
Ohya one of the possible issue is that I think the storage has no way of identifying now whether a session is a recurring session or a standalone session. I tried creating the recurring session, close the app and open it again, and the recurring session becomes a standalone session now. Might need to take the storage into consideration, either in this iteration, or as a separate issue.

Reply:
Right ive got to make a RecurringSession json. to save things into if Session instanceOf RecurringSession

Reply:
Just done so, I think should be working